src/pages/AdminView.jsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+const AdminView = () => {
+    const [user, setUser] = useState(null);
+
+    useEffect(() => {
+        const params = new URLSearchParams(location.search);
+        const user = {
+          firstName: params.get('firstName'),
+          lastName: params.get('lastName'),
+          username: params.get('username')
+        };
+        setUser(user);
+      }, [location]);
```

Comment:
nit: You don't need to put location in the useEffect dependency array. If the array is empty, the useEffect will just happen on the first initial render so the page will still get the user's name.